src/main/java/com/gojek/beast/com/gojek/beast/protomapping/ProtoUpdateListener.java

```diff
@@ -0,0 +1,96 @@
+package com.gojek.beast.com.gojek.beast.protomapping;
+
+import com.gojek.beast.Clock;
+import com.gojek.beast.config.AppConfig;
+import com.gojek.beast.config.ColumnMapping;
+import com.gojek.beast.config.ProtoMappingConfig;
+import com.gojek.beast.converter.ConsumerRecordConverter;
+import com.gojek.beast.converter.RowMapper;
+import com.gojek.beast.models.ExternalCallException;
+import com.gojek.beast.models.UpdateBQTableRequest;
+import com.gojek.de.stencil.client.StencilClient;
+import com.gojek.de.stencil.parser.ProtoParser;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+
+@Slf4j
+public class ProtoUpdateListener extends com.gojek.de.stencil.cache.ProtoUpdateListener {
+    private ConsumerRecordConverter recordConverter;
+    private ProtoMappingConfig protoMappingConfig;
+    private AppConfig appConfig;
+    private String proto;
+    private StencilClient stencilClient;
+
+    public ProtoUpdateListener(ProtoMappingConfig protoMappingConfig, AppConfig appConfig, StencilClient stencilClient, String proto) throws ExternalCallException {
+        super(proto);
+        this.proto = proto;
+        this.protoMappingConfig = protoMappingConfig;
+        this.appConfig = appConfig;
+        this.stencilClient = stencilClient;
+        this.setProtoParser(getProtoMapping());
+    }
+
+    @Override
+    public void onProtoUpdate() {
+        log.info("updating bq table as {} proto was updated", this.getProto());
+        try {
+            this.updateProtoParser();
+        } catch (ExternalCallException e) {
+            log.warn("Error while updating bigquery table:" + e.getMessage());
+        }
+    }
+
+    // First get latest protomapping, update bq schema, and if all goes fine
+    // then only update beast's proto mapping config
+    private void updateProtoParser() throws ExternalCallException {
+        String protoMapping = getProtoMappingFromRemoteURL();
+        updateBigQuerySchema();
+        protoMappingConfig.setProperty("PROTO_COLUMN_MAPPING", protoMapping);
+        setProtoParser(protoMappingConfig.getProtoColumnMapping());
+    }
+
+    private ColumnMapping getProtoMapping() throws ExternalCallException {
+        if (protoMappingConfig.getProtoColumnMappingURL() != null) {
+            String protoMapping = this.getProtoMappingFromRemoteURL();
+            protoMappingConfig.setProperty("PROTO_COLUMN_MAPPING", protoMapping);
```

Comment:
We have to handle the scenario where `getProtoMappingFromRemoteUrl` throwing Exception

Reply:
@devdinu the `getProtoMappingFromRemoteUrl` function is called twice.
1. At startup when we need to fetch proto mapping from remote url
2. When proto changes and we need to refetch the latest proto mapping.

We are handling case 2 in `onProtoUpdate` function which will be called if proto is updated.
IMO, for the first case we should not handle the error as this is non recoverable.
In first case we should propogate the error and stop the application.